pageserver/src/tenant/timeline.rs

```diff
@@ -939,6 +983,9 @@ impl Timeline {
             }
         }
 
+        let background_ctx =
+            RequestContext::new(TaskKind::WalReceiverManager, DownloadBehavior::Error);
```

Comment:
`DownloadBehavior::Error` @hlinnaka I copy-pasted this from your PR. Are you sure this works in practice? IIRC walreceiver sometimes needs to access relsize or so to determine whether rel extension is needed.

Reply:
Ingesting WAL records indeed needs to access relation sizes, but this context is for the "connection manager". A separate child context is created for each connection which performs the ingestion.